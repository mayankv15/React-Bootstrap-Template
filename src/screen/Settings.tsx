import React, { useState } from "react";
import { Typography } from "@mui/material";
import Select from "../components/common/controls/Select";
import { useTranslation } from "react-i18next";
import i18next from "i18next";
import CONSTANTS from "../constants/constants";

const Settings = () => {
  const { t: translation } = useTranslation();
  const language = sessionStorage.getItem(CONSTANTS.I18NEXTLNG) || "en";
  const country = sessionStorage.getItem(CONSTANTS.COUNTRY) || "IN";
  const [languageValue, setLanguageValue] = useState(language);
  const [countryValue, setCountryValue] = useState(country);
  const LanguageList = {
    en: { title: "english", dir: "ltr", value: "en", code: "en" },
    ta: { title: "tamil", dir: "ltr", value: "ta", code: "ta" },
  };
  const selectLanguage = Object.values(LanguageList)?.map((item) => item);
  const selectCountry = [
    { title: "india", value: "IN" },
    { title: "japan", value: "JA" },
  ];
  const handleLanguageChange = (event) => {
    const [selectedLanguage] = selectLanguage.filter(
      (item) => item.value === event.target.value
    );
    i18next.changeLanguage(selectedLanguage?.code);
    setLanguageValue(selectedLanguage?.code);
  };
  const handleCountryChange = (event) => {
    sessionStorage.setItem(CONSTANTS.COUNTRY, event.target.value);
    setCountryValue(event.target.value);
  };

  React.useEffect(() => {
    document.body.dir = LanguageList[language]?.dir;
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [language]);

  return (
    <div
      style={{
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        paddingTop: "50px",
      }}
    >
      <Typography variant="h3">
        <div
          style={{
            display: "flex",
            alignItems: "center",
            justifyItems: "center",
          }}
        >
          <Typography
            variant="h6"
            style={{ width: "300px", paddingTop: "35px" }}
          >
            {translation("select_country")}
          </Typography>
          <Select
            options={selectCountry}
            label={"           "}
            onChange={handleCountryChange}
            value={countryValue}
          />
        </div>
        <div style={{ display: "flex", justifyContent: "center" }}>
          <Typography
            variant="h6"
            style={{ width: "300px", paddingTop: "35px" }}
          >
            {translation("select_language")}
          </Typography>
          <Select
            options={selectLanguage}
            label={"           "}
            onChange={handleLanguageChange}
            value={languageValue}
          />
        </div>
      </Typography>
    </div>
  );
};

export default Settings;
