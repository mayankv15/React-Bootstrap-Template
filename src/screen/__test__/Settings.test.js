import React from "react";
import { render } from "@testing-library/react";
import Settings from "../Settings"

describe("Renders Settings", () => {
    it("should render Settings correctly", () => {
        const component = render(<Settings />);
        expect(component).toMatchSnapshot();
    });
});
