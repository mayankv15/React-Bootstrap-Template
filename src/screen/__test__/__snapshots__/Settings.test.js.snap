// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Settings should render Settings correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    dir="ltr"
  >
    <div>
      <div
        style="display: flex; justify-content: center; align-items: center; padding-top: 50px;"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3"
        >
          <div
            style="display: flex; align-items: center; justify-items: center;"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6"
              style="width: 300px; padding-top: 35px;"
            >
              select_country
            </h6>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                data-shrink="true"
                id="demo-simple-select-label"
                style="width: 110px; padding-left: 5px;"
              >
                           
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-customSelect-1 MuiInputBase-formControl MuiInput-formControl"
                style="width: 165px; padding-top: 15px;"
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="demo-simple-select-label demo-simple-select"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                  id="demo-simple-select"
                  role="button"
                  tabindex="0"
                >
                  india
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput"
                  tabindex="-1"
                  value="IN"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            style="display: flex; justify-content: center;"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6"
              style="width: 300px; padding-top: 35px;"
            >
              select_language
            </h6>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                data-shrink="true"
                id="demo-simple-select-label"
                style="width: 110px; padding-left: 5px;"
              >
                           
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-customSelect-1 MuiInputBase-formControl MuiInput-formControl"
                style="width: 165px; padding-top: 15px;"
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="demo-simple-select-label demo-simple-select"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                  id="demo-simple-select"
                  role="button"
                  tabindex="0"
                >
                  english
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput"
                  tabindex="-1"
                  value="en"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </h3>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex; justify-content: center; align-items: center; padding-top: 50px;"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3"
      >
        <div
          style="display: flex; align-items: center; justify-items: center;"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
            style="width: 300px; padding-top: 35px;"
          >
            select_country
          </h6>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink="true"
              id="demo-simple-select-label"
              style="width: 110px; padding-left: 5px;"
            >
                         
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-customSelect-1 MuiInputBase-formControl MuiInput-formControl"
              style="width: 165px; padding-top: 15px;"
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="demo-simple-select"
                role="button"
                tabindex="0"
              >
                india
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                tabindex="-1"
                value="IN"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          style="display: flex; justify-content: center;"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
            style="width: 300px; padding-top: 35px;"
          >
            select_language
          </h6>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink="true"
              id="demo-simple-select-label"
              style="width: 110px; padding-left: 5px;"
            >
                         
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-customSelect-1 MuiInputBase-formControl MuiInput-formControl"
              style="width: 165px; padding-top: 15px;"
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="demo-simple-select"
                role="button"
                tabindex="0"
              >
                english
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                tabindex="-1"
                value="en"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </h3>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
